app/history.py

```diff
@@ -19,44 +19,65 @@
 import app.log
 import cPickle as pickle
 import os
+import hashlib
 
-data = {}
-path = None
+fileHistory = {}
+userHistory = {}
+pathToHistory = None
+hasher = None
+checksum = None
+fileSize = 0
 
-def get(keyPath, default=None):
-  global data
-  cursor = data
-  for i in keyPath[:-1]:
-    cursor = cursor.setdefault(i, {})
-  return cursor.get(keyPath[-1], default)
+def get(keyPath, default={}):
+  return fileHistory.setdefault(keyPath[-1], default)
```

Comment:
Yeah I don't really know. When I first started this PR, history and bookmarks were saved with keys that were like ['files', self.fullPath, **attribute here**]. I didn't know if you had a reason, but I just kept it for now. I can drop all these and just have a string as the key instead of an array if the rest doesn't have any purpose.